src/sources/aws_s3/mod.rs

```diff
@@ -524,26 +581,83 @@ mod integration_tests {
     ) {
         let key = key.unwrap_or_else(|| uuid::Uuid::new_v4().to_string());
 
-        let s3 = s3_client();
-        let sqs = sqs_client();
+        let s3 = s3_client().await;
+        let sqs = sqs_client().await;
 
         let queue = create_queue(&sqs).await;
-        let bucket = create_bucket(&s3, &queue).await;
+        let bucket = create_bucket(&s3).await;
+
+        tokio::time::sleep(Duration::from_secs(1)).await;
 
         let config = config(&queue, multiline);
 
-        s3.put_object(PutObjectRequest {
-            bucket: bucket.clone(),
-            key: key.clone(),
-            body: Some(rusoto_core::ByteStream::from(payload)),
-            content_type: content_type.map(|t| t.to_owned()),
-            content_encoding: content_encoding.map(|t| t.to_owned()),
-            ..Default::default()
-        })
-        .await
-        .expect("Could not put object");
+        s3.put_object()
+            .bucket(bucket.clone())
+            .key(key.clone())
+            .body(ByteStream::from(payload))
+            .set_content_type(content_type.map(|t| t.to_owned()))
+            .set_content_encoding(content_encoding.map(|t| t.to_owned()))
+            .send()
+            .await
+            .expect("Could not put object");
+
+        let sqs_client = sqs_client().await;
+
+        let mut s3_event: S3Event = serde_json::from_str(
+            r#"
+{
+   "Records":[
+      {
+         "eventVersion":"2.1",
+         "eventSource":"aws:s3",
+         "awsRegion":"us-east-1",
+         "eventTime":"2022-03-24T19:43:00.548Z",
+         "eventName":"ObjectCreated:Put",
+         "userIdentity":{
+            "principalId":"AWS:ARNOTAREALIDD4:user.name"
+         },
+         "requestParameters":{
+            "sourceIPAddress":"136.56.73.213"
+         },
+         "responseElements":{
+            "x-amz-request-id":"ZX6X98Q6NM9NQTP3",
+            "x-amz-id-2":"ESLLtyT4N5cAPW+C9EXwtaeEWz6nq7eCA6txjZKlG2Q7xp2nHXQI69Od2B0PiYIbhUiX26NrpIQPV0lLI6js3nVNmYo2SWBs"
+         },
+         "s3":{
+            "s3SchemaVersion":"1.0",
+            "configurationId":"asdfasdf",
+            "bucket":{
+               "name":"bucket-name",
+               "ownerIdentity":{
+                  "principalId":"A3PEG170DF9VNQ"
+               },
+               "arn":"arn:aws:s3:::nfox-testing-vector"
+            },
+            "object":{
+               "key":"test-log.txt",
+               "size":33,
+               "eTag":"c981ce6672c4251048b0b834e334007f",
+               "sequencer":"00623CC9C47AB5634C"
+            }
+         }
+      }
+   ]
+}
+        "#,
+        )
+        .unwrap();
+
+        s3_event.records[0].s3.bucket.name = bucket.clone();
+        s3_event.records[0].s3.object.key = key.clone();
 
-        assert_eq!(count_messages(&sqs, &queue).await, 1);
+        // send SQS message (this is usually sent by S3 itself when an object is uploaded)
```

Comment:
I assume this is the localstack bug you were working around? Would it make sense to leave a comment about that?